dingo/core/models/posterior_model.py

```diff
@@ -185,6 +190,47 @@ def save_model(
 
         torch.save(model_dict, model_filename)
 
+
+    def _load_model_from_hdf5(
+        self,
+        model_filename: str
+    ):
+        """
+        Helper function to load a trained model that has been
+        saved in HDF5 format using `pt_to_hdf5.py`.
+
+        Parameters
+        ----------
+        model_filename: str
+            path to saved model; must have extension '.hdf5'
+
+        Returns
+        -------
+        d: dict
+            A stripped down version of the dict saved by torch.save()
+            Specifically, it does not include 'optimizer_state_dict'
+            to save space at inference time.
+        """
+        d = {}
+        with h5py.File(model_filename, 'r') as fp:
+            model_basename = os.path.basename(model_filename)
+            if fp.attrs['CANONICAL_FILE_BASENAME'] != model_basename:
+                raise ValueError('HDF5 attribute CANONICAL_FILE_BASENAME differs from model name',
+                        model_basename)
+
+            # Load small nested dicts from json
+            for k, v in fp['serialized_dicts'].items():
+                d[k] = json.loads(v[()])
+
+            # Load model weights
+            model_state_dict = OrderedDict()
```

Comment:
Why use an `OrderedDict` here? `dict` objects are already ordered now in Python.

Reply:
Ah, I see. The PyTorch `state_dict` object is an `OrderedDict`.

Reply:
Exactly.